src/main/java/de/wi2020sebgroup1/cinema/controller/AccountController.java

```diff
@@ -39,10 +40,27 @@
 	@PutMapping("/registration")
 	public ResponseEntity<Object> register(@RequestBody UserRegistrationObject uro){
 		
+		boolean alreadyExists = false;
+		
 		if(!(uro.passwordHash.equals(uro.passwordConfirmHash))) {
 			return new ResponseEntity<Object>("Incorrect password!", HttpStatus.UNAUTHORIZED);
 		}
 		
+		
+		try {
+			User user = userRepository.findByUsernameSpecial(uro.username);
+			alreadyExists = true;
+		}catch(Exception e) {};
+		
+		try {
+			User user = userRepository.findByEmailEquals(uro.email).get();
+			alreadyExists = true;
+		}catch(Exception e) {};
+		
+		if(alreadyExists) {
+			return new ResponseEntity<Object>(new UserAlreadyExistsException(uro.email, uro.username), HttpStatus.CONFLICT);

```

Comment:
Würde eher 406 Not Acceptable nehmen, Conflict ist eher für Anfragekonflikte technischer Art